### PR TITLE
Fix Checkpoints fuzz overflow

### PR DESCRIPTION
**Pull request by ernestognw**
There was an edge case on the property tests for `Checkpoints.sol` when checking for reverts when setting past keys. When the key is 0, substracting 1 results on an Arithmetic overflow/underflow error.

### CODE REVIEW

File: scripts/generate/templates/Checkpoints.t.js [hunk 1]
```diff
@@ -86,10 +86,13 @@ function testPush(
 
     if (keys.length > 0) {
         ${opts.keyTypeName} lastKey = keys[keys.length - 1];
-        pastKey = _bound${capitalize(opts.keyTypeName)}(pastKey, 0, lastKey - 1);
-
-        vm.expectRevert();
-        this.push(pastKey, values[keys.length % values.length]);
+        if (lastKey > 0) {
+            ${opts.keyTypeName} maxKey = lastKey == 0 ? 0 : lastKey - 1;
+            pastKey = _bound${capitalize(opts.keyTypeName)}(pastKey, 0, maxKey);
```
Comment (frangio) [inline scripts/generate/templates/Checkpoints.t.js:91]: The `maxKey` variable is redundant because of the conditional.

```suggestion
            pastKey = _bound${capitalize(opts.keyTypeName)}(pastKey, 0, lastKey - 1);
```

**Review by frangio — CHANGES_REQUESTED** [review]

**Review by Amxx — APPROVED** [review]
